### PR TITLE
Added github actions workflow

### LINKED ISSUE #318: Migrate from TravisCI to GitHub Actions
This is needed to be able to merge any pull requests, as the tests are not being run currently.

### PR DESCRIPTION
**Pull request by Anushka-shukla**
Fixes #318 Migrate from TravisCI to GitHub Actions 

## Description
 We need to migrate from Travis CI to GitHub Actions 

## Motivation and Context

 https://github.com/common-workflow-language/cwlviewer/issues/318 

## How Has This Been Tested?
 I created the GitHub actions workflow and checked it by running it on my forked repository, and it did pass all the checks.

## Screenshots (if appropriate):
Screenshot from 2021-04-14 18-53-21

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by mr-c — CHANGES_REQUESTED** [review]
So exciting, thank you!

I see an error in the logs
```
 Error:  Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.3.0:report (default-cli) on project cwlviewer: Build error: Either repository token or service with job id must be defined -> [Help 1]
```

https://github.com/common-workflow-language/cwlviewer/runs/2343500160?check_suite_focus=true#step:9:10769

**Comment by mr-c** [comment]
Can we name the workflow some different than `.github/workflows/auto-merge.yml` ? I don't think this workflow has anything to do with merging :-)

**Comment by mr-c** [comment]
Can you also delete `.travis.yml` as part of this PR?

**Comment by mr-c** [comment]
I edited the description to say "Fix #" and then the issue number so that merging this pull request will automatically close that issue :-)

File: .github/workflows/auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ 
+     - name: Build and test with Maven
+       run: |
+         ./mvnw test -B
+ 
+     - name: Coverage report
+       run: |
+         # upload coverage report
+         mvn prepare-package -DskipTests=true
+         mvn cobertura:cobertura coveralls:report -DrepoToken=${{ secrets.COVERALLS_TOKEN }}
```
Comment (mr-c) [inline .github/workflows/auto-merge.yml:50]:
```suggestion
         mvn cobertura:cobertura coveralls:report -DrepoToken=${{ secrets.COVERALLS_TOKEN }} 
```

Consider implementing the suggestions at https://github.com/trautonen/coveralls-maven-plugin/issues/136#issuecomment-797322126

**Comment by Anushka-shukla** [comment]
> Can we name the workflow some different than `.github/workflows/auto-merge.yml` ? I don't think this workflow has anything to do with merging :-)

Sure, can you suggest something?

File: .github/workflows/auto-merge.yml (hunk 1 above)
Comment (Anushka-shukla) [reply .github/workflows/auto-merge.yml:50]: Okay

**Comment by Anushka-shukla** [comment]
> So exciting, thank you!
> 
> I see an error in the logs
> 
> ```
>  Error:  Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.3.0:report (default-cli) on project cwlviewer: Build error: Either repository token or service with job id must be defined -> [Help 1]
> ```
> 
> https://github.com/common-workflow-language/cwlviewer/runs/2343500160?check_suite_focus=true#step:9:10769

I'll try fixing this.

**Comment by mr-c** [comment]
> > So exciting, thank you!
> > I see an error in the logs
> > ```
> >  Error:  Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.3.0:report (default-cli) on project cwlviewer: Build error: Either repository token or service with job id must be defined -> [Help 1]
> > ```
> > 
> > 
> > https://github.com/common-workflow-language/cwlviewer/runs/2343500160?check_suite_focus=true#step:9:10769
> 
> We have to add the Coverall authentication token to secrets of this repository, only after that, it will be able to upload the coverage report on Coveralls. More about adding secrets to here:https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository

Secrets are not available to GitHub Actions workflows running from a PR based upon a fork of the main repo where the secret is stored, except for `GITHUB_TOKEN`

> With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository.

https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

So as documented in https://github.com/trautonen/coveralls-maven-plugin/issues/136#issuecomment-797322126 you can pass the `GITHUB_TOKEN` to `-DrepoToken=`; but please do so via environment variables (as documented there) so the token isn't printed in the log.

**Comment by Anushka-shukla** [comment]
> > > So exciting, thank you!
> > > I see an error in the logs
> > > ```
> > >  Error:  Failed to execute goal org.eluder.coveralls:coveralls-maven-plugin:4.3.0:report (default-cli) on project cwlviewer: Build error: Either repository token or service with job id must be defined -> [Help 1]
> > > ```
> > > 
> > > 
> > > https://github.com/common-workflow-language/cwlviewer/runs/2343500160?check_suite_focus=true#step:9:10769
> > 
> > 
> > We have to add the Coverall authentication token to secrets of this repository, only after that, it will be able to upload the coverage report on Coveralls. More about adding secrets to here:https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository
> 
> Secrets are not available to GitHub Actions workflows running from a PR based upon a fork of the main repo where the secret is stored, except for `GITHUB_TOKEN`
> 
> > With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository.
> 
> https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow
> 
> So as documented in [trautonen/coveralls-maven-plugin#136 (comment)](https://github.com/trautonen/coveralls-maven-plugin/issues/136#issuecomment-797322126) you can pass the `GITHUB_TOKEN` to `-DrepoToken=`; but please do so via environment variables (as documented there) so the token isn't printed in the log.

Made the changes!

**Pull request merged**